task_2/src/controllers/userController.ts

```diff
@@ -1,246 +1,111 @@
+  const schemaValidation: { error: ValidationError; value: UserRequestBody } = userRequestSchema.validate(req.body)
 
-  if (error) {
+  if (schemaValidation.error) {
     return res.status(400).json({
       success: false,
-      message: error.details[0].message,
-      error,
+      message: schemaValidation.error.details[0].message,
+      error: schemaValidation.error,
     })
   }
 
   try {
-    const users = await _getUsers()
-    const isLoginNotAvailable = users.find((user) => user.login === value.login)
+    const result = await userService.create(schemaValidation.value)
 
-    if (isLoginNotAvailable) {
-      return res.status(400).json({
-        success: false,
-        message: LOGIN_NOT_AVAILABLE_MESSAGE,
-        data: value,
-      })
-    }
-
-    const user: User = {
-      id: uuidv4(),
-      login: value.login,
-      password: value.password,
-      age: value.age,
-      isDeleted: false,
+    if (result?.error) {
+      return res.status(400).json(result.error)
     }
 
-    const newUsers = jsonStringfy([...users, user])
-
-    await fs.writeFile(databaseFilePath, newUsers)
-
-    res.status(201).json({
-      success: true,
-      message: USER_CREATED_MESSAGE,
-      data: user,
-    })
+    res.status(201).json(result.response)
   } catch (error) {
-    res.status(500).json({
-      success: false,
-      message: SOMETHING_WENT_WRONG_MESSAGE,
-      error: error.message,
-    })
+    res.status(500).json(getGenericErrorMessage(error))
   }
 }
 
 export const updateUser = async (req: Request, res: Response) => {
   try {
-    // check if the user with the provided ID exists
-    const userId = req.params.id
-    const user = await _getUserById(userId)
-
-    if (!user) {
-      return res.status(404).json({
-        success: false,
-        error: NO_USER_FOUND_MESSAGE,
-        data: userId,
-      })
-    }
+    const schemaValidation: { error: ValidationError; value: UserRequestBody } = userRequestSchema.validate(req.body)
 
-    // check if all the required fields are provided
-    const { error, value }: { error: ValidationError; value: UserRequestBody } = userRequestSchema.validate(req.body)
-
-    if (error) {
+    if (schemaValidation.error) {
       return res.status(400).json({
         success: false,
-        message: error.details[0].message,
-        error,
+        message: schemaValidation.error.details[0].message,
+        error: schemaValidation.error,
       })
     }
 
-    const users = await _getUsers()
-    const isLoginNotAvailable = users.find((user) => user.login === value.login)
-
-    if (isLoginNotAvailable) {
-      return res.status(400).json({
-        success: false,
-        message: LOGIN_NOT_AVAILABLE_MESSAGE,
-        data: value,
-      })
+    const userId = parseInt(req.params.id)
+    const updatedUser: IUser = {
+      id: userId,
+      login: schemaValidation.value.login,
+      password: schemaValidation.value.password,
```

Comment:
don't forget that in real life scenarios login and password fields are not updated like this!!!
usually we will have separate endpoints for them. 

now it is not an issue since task description does not specifically mentioned it!

Reply:
Yeah, you are right. I think in the upcoming homework tasks we have proper authorization practices